packages/commonwealth/client/scripts/views/components/unified_user_profiles_flow.tsx

```diff
@@ -0,0 +1,49 @@
+/* @jsx m */
+
+import ClassComponent from 'class_component';
+import m from 'mithril';
+
+import 'modals/unified_user_flow_modal.scss';
+
+import app from 'state';
+import { UnifiedUserFlowModal } from '../modals/unified_user_flow_modal';
+
+const DISPLAY_INTERVAL = 60 * 60 * 1000; // 1 hour wait
+
+// Exists only to abstract user profile modal logic outside of Layout
+export class UnifiedUserFlow extends ClassComponent {
+  private unifyProfileModalOn = false;
+
+  view(vnode) {
+    const userHasDisplayName = app.user.hasDisplayName;
+
+    if (!this.unifyProfileModalOn && !userHasDisplayName) {
+      const lastShownProfileModal = localStorage.getItem(
+        'user-profile-modal-last-displayed'
+      );
+
+      if (!lastShownProfileModal) {
+        localStorage.setItem(
+          'user-profile-modal-last-displayed',
+          Date.now().toString()
+        );
+        app.modals.create({ modal: UnifiedUserFlowModal });
+      } else {
+        const now = new Date();
```

Comment:
So, if i'm correct here, we show the modal once an hour, regardless of whether they've seen it before / set a user name, but we show it immediately on first site load if they don't have a display name?

The logic in this component is not very clear to me...

Reply:
Logic is: 

If never seen modal before (localstorage item doesnt exist), show immediately.
Otherwise, check if the last time modal was seen was more than an hour ago. If so, show modal.

Reply:
What does the `this.unifyProfileModalOn` do?

Reply:
its preventing rerenders from re-creating the modal